### PR TITLE
Update to minimum CMake version of 3.14

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Update docker images to images with CMake 3.14+ installed.

Update global.json to request CMake 3.14.2.

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
Blocked on: dotnet/dotnet-buildtools-prereqs-docker#187

**Comment by jkoritzinsky** [comment]
cc: @dotnet/coreclr-infra

File: eng/platform-matrix-managed-test-build.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
       managedTestBuildOsGroup: ${{ parameters.armManagedTestBuildOsGroup }}
       managedTestBuildOsSubgroup: ${{ parameters.armManagedTestBuildOsSubgroup }}
       container:
-        image: ubuntu-16.04-cross-14.04-23cacb0-20190528233931
+        image: ubuntu-16.04-cross-14.04-23cacb0-20190923200213
```
Comment (jashook) [inline eng/platform-matrix-managed-test-build.yml]: Could you update the documentation to reflect these image changes?

File: global.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "dotnet": "3.0.100-preview7-012821"
   },
   "native-tools": {
-    "cmake": "3.11.1",
+    "cmake": "3.14.2",
```
Comment (ViktorHofer) [inline global.json]: Is this actually doing anything as native tools installation is disabled: https://github.com/dotnet/coreclr/blob/4e702da2c06ded8239d5eb2f47619604efbf3fde/eng/configure-toolset.ps1#L12?

Comment (jkoritzinsky) [reply global.json]: We manually call the init-native-tools.ps1 script in build.cmd and build-test.cmd

Comment (ViktorHofer) [reply global.json]: I see, thanks.

**Pull request merged**